transport/transport.go

```diff
@@ -303,6 +303,24 @@ func StreamFromContext(ctx context.Context) (s *Stream, ok bool) {
 	return
 }
 
+// Peer connection info
+type Peer struct {
```

Comment:
Doing so makes a circular dependency between the `grpc` and `grpc/transport` packages unless `gprc.Peer` is an interface (among other things.)  Happy to do this but it might be a bit kludgy.